app/controllers/provider_interface/candidate_pool/invites_controller.rb

```diff
@@ -3,15 +3,36 @@ module CandidatePool
     class InvitesController < ProviderInterfaceController
       include Pagy::Backend
       before_action :redirect_to_applications_unless_provider_opted_in
+      before_action :set_invite, only: :show
+      before_action :redirect_if_candidate_in_pool, only: :show
 
       def index
         @filter = CandidateInvitesFilter.new(filter_params:, provider_user: current_provider_user)
 
         @pagy, @candidate_invites = pagy(@filter.applied_filters)
       end
 
+      def show; end
+
     private
 
+      def set_invite
+        @invite = Pool::Invite
+                    .published
+                    .current_cycle
+                    .published
+                    .where(provider: current_provider_user.providers)
+                    .find(params[:id])
```

Comment:
```suggestion
                    .find_by(id: params[:id])
```
I think `.find` will blow up if it doesn't find anything